usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
> I'd much prefer to preserve the pad for when we really need it

That is an understandable desire, but it sounds like you guys are exhausting all reasonable and unreasonable alternatives already. The pad is there to be used for some "future feature", so maybe it is time to use it as it should be :)

A further feature which may require it, could potentially still work, as long as encryption isn't enabled. Sort of how you guys are considering encryption vs dedup. Can it be turned into a "union" of sorts to be more visible for other future features?

But at the end of the day, whatever you prefer Matt, is what we will do.